pkg/solidity-utils/contracts/math/BasePoolMath.sol

```diff
@@ -263,12 +254,11 @@ library BasePoolMath {
         // Calculate the new invariant using the new balances (after the removal).
         // Calculate the new invariant ratio by dividing the new invariant by the old invariant.
         // Calculate the new proportional balance by multiplying the new invariant ratio by the current balance.
-        // Calculate the taxable amount by subtracting the new balance from the equivalent proportional balance,
-        // rounding in favor of the protocol. We round the first term up to subtract from more and get a higher
-        // `taxableAmount`, which charges higher swap fees, augmenting the amount of BPT that will be burned.
-        uint256 taxableAmount = computeInvariant(newBalances).divUp(currentInvariant).mulUp(
-            currentBalances[tokenOutIndex]
-        ) - newBalances[tokenOutIndex];
+        // Calculate the taxable amount by subtracting the new balance from the equivalent proportional balance.
+        // Since we multiply and divide we don't need to use FP math.
+        uint256 taxableAmount = (computeInvariant(newBalances) * currentBalances[tokenOutIndex]) /
+            currentInvariant -
+            newBalances[tokenOutIndex];
```

Comment:
This one probably divided first and multiplied later to avoid overflow since an invariant may be a big number. If this is the case, maybe it's not worth changing this one.

Reply:
As discussed off-thread, the maximum intermediate value might be larger with this approach, but I don't think that's going to be a problem.

If I'm not mistaken, the existing code in `main` is a result of an optimization after the first iteration of the code, where `invariantRatio` was an intermediate value in the stack: https://github.com/balancer/balancer-v3-monorepo/pull/145/files#diff-7732d3024d950d26c8460e3d18e8b260441c8527d497da76b921a66e6477b313R225-R227. I don't think there's a deeper reason behind the previous order.

On the other hand, the order using FP math is not _that_ important, but with raw math as with this PR we must multiply first to avoid losing precision.

Reply:
We're losing the rounding direction though, right? It's going to round down instead of up. Do we need to say `(computeInvariant(newBalances) * currentBalances[tokenOutIndex] + 1) / currentInvariant`